frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-cdrom-patches.ts

```diff
@@ -0,0 +1,169 @@
+import { last, flatten, includes } from 'lodash';
+import {
+  BOOT_ORDER_SECOND,
+  BOOT_ORDER_FIRST,
+  PVC_ACCESSMODE_RWM,
+  PVC_VOLUMEMODE_FS,
+  getBootableDevicesInOrder,
+} from 'kubevirt-web-ui-components';
+import { getName } from '@console/shared';
+import { Volume, K8sResourceKind } from '@console/internal/module/k8s';
+import { WINTOOLS_CONTAINER_NAMES } from '../../../components/modals/cdrom-vm-modal/constants';
+import { CD } from '../../../types/vm';
+import { VMKind } from '../../../types';
+import {
+  getVolumes,
+  getDataVolumeTemplates,
+  getDisks,
+  getVolumeDataVolumeName,
+} from '../../../selectors/vm';
+import { StorageType } from '../../../constants/vm/storage';
+
+const getNextAvailableBootOrderIndex = (vm: VMKind) => {
+  const sortedBootableDevices = getBootableDevicesInOrder(vm);
+  const numBootableDevices = sortedBootableDevices.length;
+  const lastBootableDevice: any = last(sortedBootableDevices);
+
+  // assigned indexes start at two as the first index is assigned directly by the user
+  return numBootableDevices > 0 ? lastBootableDevice.value.bootOrder + 1 : BOOT_ORDER_SECOND;
+};
+
+const assignBootOrderIndex = (vm: VMKind, currDevBootOrder = -1) => {
+  let bootOrder = currDevBootOrder;
+  if (currDevBootOrder !== BOOT_ORDER_FIRST) {
+    bootOrder = getNextAvailableBootOrderIndex(vm);
+  }
+  return bootOrder;
+};
+
+export const getCDsPatch = (
+  vm: VMKind,
+  cds: CD[],
+  pvcs: K8sResourceKind[],
+  storageClasses: K8sResourceKind[],
+) => {
+  let newBootOrder = assignBootOrderIndex(vm);
+  let DISKS = getDisks(vm);
+  let VOLS = getVolumes(vm);
+  let DATATEMPLATES = getDataVolumeTemplates(vm);
+
+  const patches = cds
+    .filter((cd) => cd.changed)
+    .map(
+      ({ name, pvc, type, bootOrder, bus, container, windowsTools, url, storageClass, size }) => {
+        const pvcObject = pvcs.find((vol) => getName(vol) === pvc) || pvcs[0];
+        const storageClassObject =
+          storageClasses.find((storClass) => getName(storClass) === storageClass) ||
+          storageClasses[0];
+        const windowsTool = windowsTools || WINTOOLS_CONTAINER_NAMES[0];
+        const existingCD = !!bootOrder;
+
+        const disk: CD = {
+          name,
+          bootOrder: existingCD ? bootOrder : newBootOrder,
+          cdrom: { bus: bus || 'virtio' },
+        };
+        let volume: Volume = { name };
+        let dataVolumeTemplate: K8sResourceKind = null;
+
+        // Patches
+        if (type === StorageType.PVC) {
+          volume = {
+            persistentVolumeClaim: {
+              claimName: getName(pvcObject),
+            },
+            name,
+          };
+        }
+
+        if (type === StorageType.URL) {
+          dataVolumeTemplate = {
+            metadata: {
+              name: `${getName(vm)}-${name}`,
+            },
+            spec: {
+              pvc: {
+                accessModes: [PVC_ACCESSMODE_RWM],
+                volumeMode: PVC_VOLUMEMODE_FS,
```

Comment:
@mareklibra where is the config map?
only place i've seen `kubevirt-storage-class-defaults` is in test files.

Reply:
It's either in `openshift-cnv` or `openshift` namespace.

Please look for `getDefaultSCAccessMode()` method which has been migrated from web-ui-components to openshift-console recently